source/WindowScroller/WindowScroller.js

```diff
@@ -0,0 +1,76 @@
+/** @flow */
+import React, { Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+import shallowCompare from 'react-addons-shallow-compare'
+import raf from 'raf'
+
+export default class WindowScroller extends Component {
+  static propTypes = {
+    /**
+     * Function respondible for rendering children.
+     * This function should implement the following signature:
+     * ({ height, scrollTop }) => PropTypes.element
+     */
+    children: PropTypes.func.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      scrollTop: 0,
+      height: 0
+    }
+
+    this._onScroll = this._onScroll.bind(this)
+  }
+
+  componentDidMount () {
+    this._positionFromTop = ReactDOM.findDOMNode(this).getBoundingClientRect().top
+    this.setState({ height: window.innerHeight })
+    window.addEventListener('scroll', this._onScroll, false)
```

Comment:
We also need to watch for changes to `window.innerHeight`. If you start with a very small height and then it grows larger (because a user resizes) then it currently leaves the last few rows of the `Grid` empty.